ott/core/continuous_barycenter.py

```diff
@@ -129,13 +127,14 @@ def solve_linear_ot(a, x, b, y):
     else:
       errors = None
 
-    divide_a = jnp.where(self.a > 0, 1.0 / self.a, 1.0)
-    convex_weights = matrices * divide_a[None, :, None]
-    x_new = jnp.sum(
-        barycentric_projection(convex_weights, segmented_y, bar_prob.cost_fn) *
-        bar_prob.weights[:, None, None],
-        axis=0
+    # compute each point of the barycenter as the barycenter of the bar_prob.weights.shape[0] barycenters. TODO: compute directly minimizer of double sum.
+    x_new = jax.vmap(
+        bar_prob.cost_fn.barycenter, in_axes=[None, 1]
+    )(
+        bar_prob.weights,
```

Comment:
I think it could make sense to call barycentric_projection(...) below something like barycenters_per_measure and then apply the vmap on top of this, this would make this easier to parse.